Regression of encoding via intersection

tested with version 1.8.0 (not present in 1.7.1)

This should work

```typescript
    const X = t.intersection([
      t.type({
        a: tTypes.createOptionFromNullable(t.string)
      }),
      t.type({})
    ])
    expect(X.encode({ a: some('1') })).to.deep.equals({ a: '1' }) 
```

(`createOptionFromNullable` coming from `io-ts-types`)
Problem is that inner a is not encoded as part of the output:

```bash
AssertionError: expected { a: some("1") } to deeply equal { a: '1' }
      + expected - actual

       {
      -  "a": {
      -    "_tag": "Some"
      -    "value": "1"
      -  }
      +  "a": "1"
       }
```

Note: I've checked that I don't have different versions of `fp-ts`, `io-ts` and `io-ts-types` installed.

--- a/src/index.ts
+++ b/src/index.ts
@@ -1192,12 +1192,21 @@ export interface IntersectionC<CS extends [Mixed, Mixed, ...Array<Mixed>]>
     unknown
   > {}
 
-const mergeAll = (us: Array<unknown>): any => {
-  let r: unknown = us[0]
-  for (let i = 1; i < us.length; i++) {
+const mergeAll = (base: any, us: Array<any>): any => {
+  let r: any = base
+  for (let i = 0; i < us.length; i++) {
     const u = us[i]
-    if (u !== r) {
-      r = Object.assign(r, u)
+    if (u !== base) {
+      // `u` contains a prismatic value or is the result of a stripping combinator
+      if (r === base) {
+        r = Object.assign({}, u)
+        continue
+      }
+      for (const k in u) {
+        if (u[k] !== base[k] || !r.hasOwnProperty(k)) {
+          r[k] = u[k]
+        }
+      }
     }
   }
   return r
@@ -1241,9 +1250,9 @@ export function intersection<CS extends [Mixed, Mixed, ...Array<Mixed>]>(
               us.push(validation.value)
             }
           }
-          return errors.length > 0 ? failures(errors) : success(mergeAll(us))
+          return errors.length > 0 ? failures(errors) : success(mergeAll(u, us))
         },
-    codecs.length === 0 ? identity : a => mergeAll(codecs.map(codec => codec.encode(a))),
+    codecs.length === 0 ? identity : a => mergeAll(a, codecs.map(codec => codec.encode(a))),
     codecs
   )
 }

--- a/test/PathReporter.ts
+++ b/test/PathReporter.ts
@@ -8,7 +8,9 @@ describe('PathReporter', () => {
     assert.deepStrictEqual(PathReporter.report(t.number.decode(function() {})), [
       'Invalid value <function0> supplied to : number'
     ])
-    assert.deepStrictEqual(PathReporter.report(t.number.decode(function f() {})), ['Invalid value f supplied to : number'])
+    assert.deepStrictEqual(PathReporter.report(t.number.decode(function f() {})), [
+      'Invalid value f supplied to : number'
+    ])
   })
 
   it('should say something whene there are no errors', () => {

--- a/test/PathReporter.ts
+++ b/test/PathReporter.ts
@@ -5,17 +5,17 @@ import { NumberFromString } from './helpers'
 
 describe('PathReporter', () => {
   it('should use the function name as error message', () => {
-    assert.deepEqual(PathReporter.report(t.number.decode(function() {})), [
+    assert.deepStrictEqual(PathReporter.report(t.number.decode(function() {})), [
       'Invalid value <function0> supplied to : number'
     ])
-    assert.deepEqual(PathReporter.report(t.number.decode(function f() {})), ['Invalid value f supplied to : number'])
+    assert.deepStrictEqual(PathReporter.report(t.number.decode(function f() {})), ['Invalid value f supplied to : number'])
   })
 
   it('should say something whene there are no errors', () => {
-    assert.deepEqual(PathReporter.report(t.number.decode(1)), ['No errors!'])
+    assert.deepStrictEqual(PathReporter.report(t.number.decode(1)), ['No errors!'])
   })
 
   it('should account for the optional message field', () => {
-    assert.deepEqual(PathReporter.report(NumberFromString.decode('a')), ['cannot parse to a number'])
+    assert.deepStrictEqual(PathReporter.report(NumberFromString.decode('a')), ['cannot parse to a number'])
   })
 })

--- a/test/TypeClass.ts
+++ b/test/TypeClass.ts
@@ -39,13 +39,13 @@ describe('Type', () => {
 
     it('accept to pipe a type with a wider input', () => {
       const T = t.string.pipe(t.string)
-      assert.deepEqual(T.decode('a'), right('a'))
+      assert.deepStrictEqual(T.decode('a'), right('a'))
       assert.strictEqual(T.encode('a'), 'a')
     })
 
     it('accept to pipe a type with a narrower output', () => {
       const T = t.string.pipe(t.literal('foo'))
-      assert.deepEqual(T.decode('foo'), right('foo'))
+      assert.deepStrictEqual(T.decode('foo'), right('foo'))
       assert.strictEqual(T.encode('foo'), 'foo')
     })
   })
@@ -65,7 +65,7 @@ describe('Type', () => {
 
 describe('getContextEntry', () => {
   it('should return a ContextEntry', () => {
-    assert.deepEqual(t.getContextEntry('key', t.string), {
+    assert.deepStrictEqual(t.getContextEntry('key', t.string), {
       key: 'key',
       type: t.string
     })

--- a/test/array.ts
+++ b/test/array.ts
@@ -53,7 +53,7 @@ describe('array', () => {
   describe('encode', () => {
     it('should encode a isomorphic value', () => {
       const T = t.array(t.number)
-      assert.deepEqual(T.encode([1, 2, 3]), [1, 2, 3])
+      assert.deepStrictEqual(T.encode([1, 2, 3]), [1, 2, 3])
     })
 
     it('should return the same reference while encoding isomorphic values', () => {
@@ -64,7 +64,7 @@ describe('array', () => {
 
     it('should encode a prismatic value', () => {
       const T = t.array(NumberFromString)
-      assert.deepEqual(T.encode([1, 2, 3]), ['1', '2', '3'])
+      assert.deepStrictEqual(T.encode([1, 2, 3]), ['1', '2', '3'])
     })
   })
 

--- a/test/brand.ts
+++ b/test/brand.ts
@@ -33,7 +33,7 @@ interface MyNumberFromStringArrayBrand {
 }
 
 const MyNumberFromStringArray = t.brand(
-  t.array(t.number),
+  t.array(NumberFromString),
   (_): _ is t.Branded<Array<number>, MyNumberFromStringArrayBrand> => true,
   'MyNumberFromStringArray'
 )
@@ -96,7 +96,7 @@ describe('brand', () => {
   describe('encode', () => {
     it('should encode a prismatic value', () => {
       const T = MyNumberFromStringArray
-      assert.deepEqual(T.encode([1] as any), ['1'])
+      assert.deepStrictEqual(T.encode([1] as any), ['1'])
     })
 
     it('should return the same reference while encoding', () => {

--- a/test/exact.ts
+++ b/test/exact.ts
@@ -130,7 +130,7 @@ describe('exact', () => {
   describe('encode', () => {
     it('should encode a prismatic value', () => {
       const T = t.exact(t.type({ a: NumberFromString }))
-      assert.deepEqual(T.encode({ a: 1 }), { a: '1' })
+      assert.deepStrictEqual(T.encode({ a: 1 }), { a: '1' })
     })
 
     it('should return the same reference while encoding', () => {
@@ -142,7 +142,7 @@ describe('exact', () => {
     it('should strip additional properties', () => {
       const T = t.exact(t.type({ a: t.string }))
       const x = { a: 'a', b: 1 }
-      assert.deepEqual(T.encode(x), { a: 'a' })
+      assert.deepStrictEqual(T.encode(x), { a: 'a' })
     })
   })
 })

--- a/test/getIndexRecord.ts
+++ b/test/getIndexRecord.ts
@@ -9,7 +9,7 @@ describe('getIndexRecord', () => {
   interface IndexRecord extends Record<string, Array<[unknown, t.Any, t.Any]>> {}
 
   const assertEqualIndexRecord = (codecs: Array<t.Mixed>, expected: IndexRecord) => {
-    assert.deepEqual(t.getIndexRecord(codecs), expected)
+    assert.deepStrictEqual(t.getIndexRecord(codecs), expected)
   }
 
   it('should handle zero types', () => {

--- a/test/helpers.ts
+++ b/test/helpers.ts
@@ -5,7 +5,7 @@ import { PathReporter } from '../src/PathReporter'
 
 export function assertStrictEqual<T>(result: t.Validation<T>, expected: any): void {
   if (result.isRight()) {
-    assert.deepEqual(result.value, expected)
+    assert.deepStrictEqual(result.value, expected)
   } else {
     throw new Error(`${result} is not a right`)
   }
@@ -14,7 +14,7 @@ export function assertStrictEqual<T>(result: t.Validation<T>, expected: any): vo
 export function assertSuccess<T>(result: t.Validation<T>, expected?: T): void {
   if (result.isRight()) {
     if (expected !== undefined) {
-      assert.deepEqual(result.value, expected)
+      assert.deepStrictEqual(result.value, expected)
     }
   } else {
     throw new Error(`${result} is not a right`)
@@ -34,7 +34,7 @@ export function assertStrictSuccess<T>(result: t.Validation<T>, expected: T): vo
 export function assertFailure(codec: t.Any, value: unknown, errors: Array<string>): void {
   const result = codec.decode(value)
   if (result.isLeft()) {
-    assert.deepEqual(PathReporter.report(result), errors)
+    assert.deepStrictEqual(PathReporter.report(result), errors)
   } else {
     throw new Error(`${result} is not a left`)
   }

--- a/test/intersection.ts
+++ b/test/intersection.ts
@@ -47,8 +47,20 @@ describe('intersection', () => {
     })
 
     it('should decode a prismatic value', () => {
-      const T = t.intersection([t.type({ a: t.string }), t.type({ b: NumberFromString })])
-      assertSuccess(T.decode({ a: 'a', b: '1' }), { a: 'a', b: 1 })
+      const T1 = t.intersection([t.type({ a: t.string }), t.type({ b: NumberFromString })])
+      assertSuccess(T1.decode({ a: 'a', b: '1' }), { a: 'a', b: 1 })
+      const T2 = t.intersection([t.type({ b: NumberFromString }), t.type({ a: t.string })])
+      assertSuccess(T2.decode({ a: 'a', b: '1' }), { a: 'a', b: 1 })
+      const T3 = t.intersection([t.type({ b: NumberFromString }), t.type({ a: t.string }), t.type({ c: t.string })])
+      assertSuccess(T3.decode({ a: 'a', b: '1', c: 'c' }), { a: 'a', b: 1, c: 'c' })
+      const T4 = t.intersection([
+        t.type({ b: NumberFromString }),
+        t.type({ a: t.string }),
+        t.type({ c: NumberFromString })
+      ])
+      assertSuccess(T4.decode({ a: 'a', b: '1', c: '2' }), { a: 'a', b: 1, c: 2 })
+      const T5 = t.intersection([t.type({ b: NumberFromString }), t.type({})])
+      assertSuccess(T5.decode({ b: '1' }), { b: 1 })
     })
 
     it('should fail decoding an invalid value', () => {

--- a/test/intersection.ts
+++ b/test/intersection.ts
@@ -113,8 +113,20 @@ describe('intersection', () => {
     })
 
     it('should encode a prismatic value', () => {
-      const T = t.intersection([t.type({ a: t.string }), t.type({ b: NumberFromString })])
-      assert.deepStrictEqual(T.encode({ a: 'a', b: 1 }), { a: 'a', b: '1' })
+      const T1 = t.intersection([t.type({ a: t.string }), t.type({ b: NumberFromString })])
+      assert.deepStrictEqual(T1.encode({ a: 'a', b: 1 }), { a: 'a', b: '1' })
+      const T2 = t.intersection([t.type({ b: NumberFromString }), t.type({ a: t.string })])
+      assert.deepStrictEqual(T2.encode({ a: 'a', b: 1 }), { a: 'a', b: '1' })
+      const T3 = t.intersection([t.type({ b: NumberFromString }), t.type({ a: t.string }), t.type({ c: t.string })])
+      assert.deepStrictEqual(T3.encode({ a: 'a', b: 1, c: 'c' }), { a: 'a', b: '1', c: 'c' })
+      const T4 = t.intersection([
+        t.type({ b: NumberFromString }),
+        t.type({ a: t.string }),
+        t.type({ c: NumberFromString })
+      ])
+      assert.deepStrictEqual(T4.encode({ a: 'a', b: 1, c: 2 }), { a: 'a', b: '1', c: '2' })
+      const T5 = t.intersection([t.type({ b: NumberFromString }), t.type({})])
+      assert.deepStrictEqual(T5.encode({ b: 1 }), { b: '1' })
     })
 
     it('should return the same reference while encoding', () => {

--- a/test/intersection.ts
+++ b/test/intersection.ts
@@ -97,12 +97,12 @@ describe('intersection', () => {
   describe('encode', () => {
     it('should encode a isomorphic value', () => {
       const T = t.intersection([t.type({ a: t.string }), t.type({ b: t.number })])
-      assert.deepEqual(T.encode({ a: 'a', b: 1 }), { a: 'a', b: 1 })
+      assert.deepStrictEqual(T.encode({ a: 'a', b: 1 }), { a: 'a', b: 1 })
     })
 
     it('should encode a prismatic value', () => {
       const T = t.intersection([t.type({ a: t.string }), t.type({ b: NumberFromString })])
-      assert.deepEqual(T.encode({ a: 'a', b: 1 }), { a: 'a', b: '1' })
+      assert.deepStrictEqual(T.encode({ a: 'a', b: 1 }), { a: 'a', b: '1' })
     })
 
     it('should return the same reference while encoding', () => {
@@ -115,9 +115,9 @@ describe('intersection', () => {
       const A = t.exact(t.type({ a: t.string }))
       const B = t.exact(t.type({ b: t.number }))
       const T = t.intersection([A, B])
-      assert.deepEqual(T.encode({ a: 'a', b: 1 }), { a: 'a', b: 1 })
+      assert.deepStrictEqual(T.encode({ a: 'a', b: 1 }), { a: 'a', b: 1 })
       const x = { a: 'a', b: 1, c: true }
-      assert.deepEqual(T.encode(x), { a: 'a', b: 1 })
+      assert.deepStrictEqual(T.encode(x), { a: 'a', b: 1 })
     })
   })
 

--- a/test/partial.ts
+++ b/test/partial.ts
@@ -51,16 +51,16 @@ describe('partial', () => {
   describe('encode', () => {
     it('should encode a isomorphic value', () => {
       const T = t.partial({ a: t.number })
-      assert.deepEqual(T.encode({}), {})
-      assert.deepEqual(T.encode({ a: undefined }), { a: undefined })
-      assert.deepEqual(T.encode({ a: 1 }), { a: 1 })
+      assert.deepStrictEqual(T.encode({}), {})
+      assert.deepStrictEqual(T.encode({ a: undefined }), { a: undefined })
+      assert.deepStrictEqual(T.encode({ a: 1 }), { a: 1 })
     })
 
     it('should encode a prismatic value', () => {
       const T = t.partial({ a: NumberFromString })
-      assert.deepEqual(T.encode({}), {})
-      assert.deepEqual(T.encode({ a: undefined }), { a: undefined })
-      assert.deepEqual(T.encode({ a: 1 }), { a: '1' })
+      assert.deepStrictEqual(T.encode({}), {})
+      assert.deepStrictEqual(T.encode({ a: undefined }), { a: undefined })
+      assert.deepStrictEqual(T.encode({ a: 1 }), { a: '1' })
     })
   })
 
@@ -70,7 +70,7 @@ describe('partial', () => {
     assertStrictEqual(T.decode(input1), input1)
     const input2 = { a: undefined }
     assertStrictEqual(T.decode(input2), input2)
-    assert.deepEqual(T.decode({ b: 1 }).value, { b: 1 })
+    assert.deepStrictEqual(T.decode({ b: 1 }).value, { b: 1 })
     const input3 = { a: undefined, b: 1 }
     assertStrictEqual(T.decode(input3), input3)
   })
@@ -96,6 +96,6 @@ describe('partial', () => {
   it('should preserve additional properties while encoding', () => {
     const T = t.partial({ a: NumberFromString })
     const x = { a: 1, b: 'foo' }
-    assert.deepEqual(T.encode(x), { a: '1', b: 'foo' })
+    assert.deepStrictEqual(T.encode(x), { a: '1', b: 'foo' })
   })
 })

--- a/test/readonly.ts
+++ b/test/readonly.ts
@@ -59,7 +59,7 @@ describe('readonly', () => {
   describe('encode', () => {
     it('should encode a prismatic value', () => {
       const T = t.readonly(t.type({ a: NumberFromString }))
-      assert.deepEqual(T.encode({ a: 1 }), { a: '1' })
+      assert.deepStrictEqual(T.encode({ a: 1 }), { a: '1' })
     })
 
     it('should return the same reference when serializing', () => {

--- a/test/readonlyArray.ts
+++ b/test/readonlyArray.ts
@@ -61,7 +61,7 @@ describe('readonlyArray', () => {
   describe('encode', () => {
     it('should encode a prismatic value', () => {
       const T = t.readonlyArray(NumberFromString)
-      assert.deepEqual(T.encode([0, 1]), ['0', '1'])
+      assert.deepStrictEqual(T.encode([0, 1]), ['0', '1'])
     })
 
     it('should return the same reference when serializing', () => {

--- a/test/record.ts
+++ b/test/record.ts
@@ -135,7 +135,7 @@ describe('record', () => {
   describe('encode', () => {
     it('should encode a isomorphic value', () => {
       const T = t.record(t.string, t.number)
-      assert.deepEqual(T.encode({ a: 1 }), { a: 1 })
+      assert.deepStrictEqual(T.encode({ a: 1 }), { a: 1 })
     })
 
     it('should return the same reference while decoding a isomorphic value', () => {
@@ -146,12 +146,12 @@ describe('record', () => {
 
     it('should encode a prismatic value', () => {
       const T = t.record(t.string, NumberFromString)
-      assert.deepEqual(T.encode({ a: 1 }), { a: '1' })
+      assert.deepStrictEqual(T.encode({ a: 1 }), { a: '1' })
     })
 
     it('should encode a prismatic key', () => {
       const T = t.record(HyphenatedString, t.number)
-      assert.deepEqual(T.encode({ 'a-b': 1 }), { ab: 1 })
+      assert.deepStrictEqual(T.encode({ 'a-b': 1 }), { ab: 1 })
     })
 
     it('should accept an array if the codomain is `unknown`', () => {

--- a/test/recursion.ts
+++ b/test/recursion.ts
@@ -98,8 +98,8 @@ describe('recursion', () => {
           b: t.union([self, t.null])
         })
       )
-      assert.deepEqual(T.encode({ a: 0, b: null }), { a: '0', b: null })
-      assert.deepEqual(T.encode({ a: 0, b: { a: 1, b: null } }), { a: '0', b: { a: '1', b: null } })
+      assert.deepStrictEqual(T.encode({ a: 0, b: null }), { a: '0', b: null })
+      assert.deepStrictEqual(T.encode({ a: 0, b: { a: 1, b: null } }), { a: '0', b: { a: '1', b: null } })
     })
   })
 

--- a/test/refinement.ts
+++ b/test/refinement.ts
@@ -60,7 +60,7 @@ describe('refinement', () => {
   describe('encode', () => {
     it('should encode a prismatic value', () => {
       const T = t.refinement(t.array(NumberFromString), () => true)
-      assert.deepEqual(T.encode([1]), ['1'])
+      assert.deepStrictEqual(T.encode([1]), ['1'])
     })
 
     it('should return the same reference while encoding', () => {

--- a/test/strict.ts
+++ b/test/strict.ts
@@ -62,7 +62,7 @@ describe('strict', () => {
   describe('encode', () => {
     it('should encode a prismatic value', () => {
       const T = t.strict({ a: NumberFromString })
-      assert.deepEqual(T.encode({ a: 1 }), { a: '1' })
+      assert.deepStrictEqual(T.encode({ a: 1 }), { a: '1' })
     })
 
     it('should return the same reference while encoding', () => {

--- a/test/strictInterfaceWithOptionals.ts
+++ b/test/strictInterfaceWithOptionals.ts
@@ -42,7 +42,7 @@ describe('strictInterfaceWithOptionals', () => {
 
   it('should serialize a deserialized', () => {
     const T = strictInterfaceWithOptionals({ foo: NumberFromString }, { bar: t.string }, 'T')
-    assert.deepEqual(T.encode({ foo: 1 }), { foo: '1' })
+    assert.deepStrictEqual(T.encode({ foo: 1 }), { foo: '1' })
   })
 
   it('should type guard', () => {

--- a/test/taggedUnion.ts
+++ b/test/taggedUnion.ts
@@ -165,11 +165,11 @@ describe('taggedUnion', () => {
 
   describe('encode', () => {
     it('should encode a isomorphic value', () => {
-      assert.deepEqual(OptionNumber.encode({ type: 'Some', value: 1 }), { type: 'Some', value: 1 })
+      assert.deepStrictEqual(OptionNumber.encode({ type: 'Some', value: 1 }), { type: 'Some', value: 1 })
     })
 
     it('should encode a prismatic value', () => {
-      assert.deepEqual(OptionNumberFromString.encode({ type: 'Some', value: 1 }), { type: 'Some', value: '1' })
+      assert.deepStrictEqual(OptionNumberFromString.encode({ type: 'Some', value: 1 }), { type: 'Some', value: '1' })
     })
   })
 
@@ -209,7 +209,7 @@ describe('taggedUnion', () => {
 
       console.warn = warn
 
-      assert.deepEqual(log, ['[io-ts] Cannot build a tagged union for (B | A), returning a de-optimized union'])
+      assert.deepStrictEqual(log, ['[io-ts] Cannot build a tagged union for (B | A), returning a de-optimized union'])
     })
   })
 
@@ -219,7 +219,7 @@ describe('taggedUnion', () => {
     console.warn = (message: string) => log.push(message)
     t.taggedUnion('type', [t.type({ type: t.literal('a') }), t.type({ bad: t.literal('b') })])
     console.warn = original
-    assert.deepEqual(log, [
+    assert.deepStrictEqual(log, [
       '[io-ts] Cannot build a tagged union for ({ type: "a" } | { bad: "b" }), returning a de-optimized union'
     ])
   })

--- a/test/tuple.ts
+++ b/test/tuple.ts
@@ -84,12 +84,12 @@ describe('tuple', () => {
   describe('encode', () => {
     it('should encode a isomorphic value', () => {
       const T = t.tuple([t.number, t.string])
-      assert.deepEqual(T.encode([1, 'a']), [1, 'a'])
+      assert.deepStrictEqual(T.encode([1, 'a']), [1, 'a'])
     })
 
     it('should encode a prismatic value', () => {
       const T = t.tuple([NumberFromString, t.string])
-      assert.deepEqual(T.encode([1, 'a']), ['1', 'a'])
+      assert.deepStrictEqual(T.encode([1, 'a']), ['1', 'a'])
     })
 
     it('should return the same reference while encoding', () => {

--- a/test/type.ts
+++ b/test/type.ts
@@ -51,7 +51,7 @@ describe('type', () => {
       assertSuccess(T2.decode({}), { a: undefined })
 
       const T3 = t.type({ a: t.any })
-      assert.deepEqual(T3.decode({}).value, { a: undefined })
+      assert.deepStrictEqual(T3.decode({}).value, { a: undefined })
     })
 
     it('should fail decoding an invalid value', () => {
@@ -70,12 +70,12 @@ describe('type', () => {
   describe('encode', () => {
     it('should encode a isomorphic value', () => {
       const T = t.type({ a: t.string })
-      assert.deepEqual(T.encode({ a: 'a' }), { a: 'a' })
+      assert.deepStrictEqual(T.encode({ a: 'a' }), { a: 'a' })
     })
 
     it('should encode a prismatic value', () => {
       const T = t.type({ a: NumberFromString })
-      assert.deepEqual(T.encode({ a: 1 }), { a: '1' })
+      assert.deepStrictEqual(T.encode({ a: 1 }), { a: '1' })
     })
   })
 
@@ -83,7 +83,7 @@ describe('type', () => {
     const T = t.type({ a: t.string })
     const validation = T.decode({ a: 's', b: 1 })
     if (validation.isRight()) {
-      assert.deepEqual(validation.value, { a: 's', b: 1 })
+      assert.deepStrictEqual(validation.value, { a: 's', b: 1 })
     } else {
       assert.ok(false)
     }

--- a/test/union.ts
+++ b/test/union.ts
@@ -76,7 +76,7 @@ describe('union', () => {
   describe('encode', () => {
     it('should encode a prismatic value', () => {
       const T1 = t.union([t.type({ a: NumberFromString }), t.number])
-      assert.deepEqual(T1.encode({ a: 1 }), { a: '1' })
+      assert.deepStrictEqual(T1.encode({ a: 1 }), { a: '1' })
       assert.strictEqual(T1.encode(1), 1)
     })
 
@@ -93,7 +93,7 @@ describe('union', () => {
     it('should play well with stripping combinators', () => {
       const T = t.union([t.strict({ a: t.number }), t.type({ b: NumberFromString })])
       const x = { a: 1, c: true }
-      assert.deepEqual(T.encode(x), x)
+      assert.deepStrictEqual(T.encode(x), x)
     })
   })
 

